src/components/title/show/package-filter-modal/package-filter-modal.js

```diff
@@ -50,13 +63,34 @@ const PackageFilterModal = ({
   const handleReset = () => {
     onSubmit([]);
     setSelectedOptions([]);
+    setSearchFilters({});
     toggleModal();
   };
 
   const handleSubmit = () => {
-    const newSelectedPackages = allPackages.filter(({ id: packageId }) => selectedOptions.some((({ value: optionId }) => packageId === optionId)));
-
-    onSubmit(newSelectedPackages);
+    const filterBySelectedStatus = ({ isSelected, providerId }) => {
+      if (!searchFilters.selected) {
+        return true;
+      }
+
+      return searchFilters.selected === selectionStatusFilterOptions.EBSCO
+        ? isSelected && providerId === ebscoProviderId
+        : searchFilters.selected === isSelected.toString();
+    };
+
+    const newSelectedPackages = allPackages
+      .filter(
+        ({ id: packageId }) => (
+          selectedOptions.length
+            ? selectedOptions.some((({ value: optionId }) => packageId === optionId))
+            : true
```

Comment:
Please, try this way
```suggestion
          !selectedOptions.length || selectedOptions.some((({ value: optionId }) => packageId === optionId))
```